console/console-init/ui/mock-console-server/schema.js

```diff
@@ -669,6 +669,12 @@ const typeDefs = gql`
       deviceId: String!
       jsonData: String!
     ): Boolean
+    setGatewaysForDevice(
+      iotproject: ObjectMeta_v1_Input!
+      deviceId: String!
+      gateways: [String]!
```

Comment:
If I'm not wrong you need to add mandatory check in array as well, gateways and gatewayGroups should be 
gateways: [String!]!
gatewayGroups: [String!]!

@jbtrystram  wdyt?

Reply:
Is it really much more complicated to use the updateDevice mutation for that ?

I am OK with adding that mutation if that helps you, as it does not imply a transaction. That said I would suggest to add support for gateways groups as well.
For example : 
``` 
setGatewaysForDevice(
      iotproject: ObjectMeta_v1_Input!
      deviceId: String!
      via: [String]!
      viaGroup: [String]!
)
```  
@ajaypratap003 the arrays can be empty if there is no gateway :)